src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -556,6 +558,9 @@ internal async Task<ReplicationSlotOptions> CreateReplicationSlot(string command
         }
         finally
         {
+            if (columnStream != null && !bypassingStream && !_replicationCancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
Am I right in understanding that there wasn't actually a bug here, just a matter of "all disposables must be disposed"?

If so, then honestly I'd consider just not disposing rather than going through these subtle checks... ColumnStream is our own internal class, and we know whether we actually need its dispose logic to run (which I don't think we do in this case). In other words, proper disposing may add more complexity than not disposing in this specific, internal case.

But am also OK with merging it if you guys prefer that.

Reply:
> Am I right in understanding that there wasn't actually a bug here

Yes.

Reply:
Up to you - am OK either way. If we don't merge this, we could instead add a comment to ColumnStream.Dispose pointing out that it's not always disposed etc.